### PR TITLE
enable indexing with GoLand latest in prebuild as well

### PR DESCRIPTION
**Pull request by akosyakov**
## Description

This PR:
- separates system directories for stable and latest versions to allow independent indexing
- enable GoLand indexing for stable and latest versions in this repo

## Related Issue(s)

Related to https://github.com/gitpod-io/gitpod/issues/6740

### Stable
- Enable GoLand in preferences: https://ak-goland-e67ebe3338.staging.gitpod-dev.com/preferences
- Start a workspace on a prebuild branch for stable and latest: https://ak-goland-e67ebe3338.staging.gitpod-dev.com/#https://github.com/akosyakov/gitpod/tree/ak/test2
- Check that GoLand is ready to go without indexing notifications

### Latest
- Switch to `Latest Release (Unstable)` in preferences: https://ak-goland-e67ebe3338.staging.gitpod-dev.com/preferences
- Start a workspace on a prebuild branch for stable and latest: https://ak-goland-e67ebe3338.staging.gitpod-dev.com/#https://github.com/akosyakov/gitpod/tree/ak/test2
- Check that GoLand is ready to go without indexing notifications

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
@akosyakov: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: scripts/goland-latest-indexing.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+# skip indexing in regular workspaces
+if [ ! "$GITPOD_HEADLESS" = "true" ] ; then exit ; fi
+
+# resolve latest JetBrains backend version running by Gitpod
+PRODUCT_CODE=GO
+curl -sSL "https://data.services.jetbrains.com/products?code=$PRODUCT_CODE&fields=distributions%2Clink%2Cname%2Creleases&_=$(date +%s)000" > /tmp/jb_products.xml
+JB_BACKEND_DOWNLOAD_URL=$(jq -r '.[0].releases[0].downloads.linux.link' /tmp/jb_products.xml)
+echo "$JB_BACKEND_DOWNLOAD_URL"
```
Comment (felladrin) [inline scripts/goland-latest-indexing.sh:10]: We could possibly use `https://download.jetbrains.com/product?distribution=linux&code=$PRODUCT_CODE` here too. [[1](https://github.com/gitpod-io/gitpod/pull/9268)]

Comment (akosyakov) [reply scripts/goland-latest-indexing.sh:10]: Actually I'm not sure. It seems to have another semantic.

**Review by felladrin — APPROVED** [review]
Code reviewed and tested! ✅

No more indexing for Go files when opening Gitpod repo on latest version of Goland 🎉

**Pull request merged**